robotoff/insights/annotate.py

```diff
@@ -125,8 +125,15 @@ def _annotate(
         else:
             annotation_result = SAVED_ANNOTATION_RESULT
 
-        insight.annotated_result = annotation_result.status_code
-        insight.save()
+        if annotation_result.status_code in (
+            AnnotationStatus.saved.value,
+            AnnotationStatus.updated.value,
+            AnnotationStatus.error_invalid_image.value,
+            AnnotationStatus.error_missing_product.value,
+            AnnotationStatus.error_updated_product.value,
```

Comment:
Style: you could have made a "db_status" attribute to AnnotationStatus, to have here a simpler:   `annotation_result.status_code in AnnotationStatus.db_status`